crates/core/tedge_mapper/src/az/converter.rs

```diff
@@ -44,13 +47,29 @@ impl Converter for AzureConverter {
     }
 
     async fn try_convert(&mut self, input: &Message) -> Result<Vec<Message>, Self::Error> {
+        let maybe_child_id = get_child_id_from_measurement_topic(&input.topic.name)?;
+
+        let az_out_topic = match maybe_child_id {
+            Some(child_id) => Topic::new_unchecked(&format!("az/messages/events/$.sub={child_id}")),
```

Comment:
I'm not expert with Azure topic schemes, so I took a quick look in their docs. The only topic scheme that I could find was `devices/{device-id}/messages/events/{property_bag}` where the `device-id` comes earlier in the topic. I'm assuming that it is for the main device and this `&.sub` property_bag scheme is specifically for child devices. But, I couldn't find any documentation on the same to confirm. So, could you share where this is documented?

Reply:
I didn't find it in the documentation directly. I only found a hint in the [plug and play mechanism](https://learn.microsoft.com/en-us/azure/iot-develop/tutorial-multiple-components?pivots=programming-language-ansi-c).
After that I discovered the [plug and play bridge](https://github.com/Azure/iot-plug-and-play-bridge) sources and found this [line](https://github.com/Azure/iot-plug-and-play-bridge/blob/master/pnpbridge/src/pnpbridge/common/pnp_protocol.c#L115), where the component name is added as property to the message.

Reply:
I posted a link in the original ticket [1799](https://github.com/thin-edge/thin-edge.io/issues/1799#issuecomment-1461538063), which shows an [example](https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/iot-central-connect-device-c.md#run-the-code) of using it (a bit buried in the log output though)

Reply:
I found some additional information in the [docs](https://learn.microsoft.com/en-us/azure/iot-develop/concepts-convention#telemetry)